android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt

```diff
@@ -257,18 +257,25 @@ private fun Answer(type: QuestionDisplayType, onAnswersClick: (List<Int>) -> Uni
             ),
             highlightStyle = EmojiHighlightStyle.ONE,
             onIndexChange = {
-                onAnswersClick(listOf(it))
+                onAnswersChange(listOf(AnswerInput.Index(it)))
             }
         )
         is QuestionDisplayType.Nps -> NpsAnswer(onIndexChange = {
-            onAnswersClick(listOf(it))
+            onAnswersChange(listOf(AnswerInput.Index(it)))
         })
         is QuestionDisplayType.Choice -> MultipleChoicesAnswer(
             choices = type.answers,
             onIndexesChange = {
-                onAnswersClick(it.toList())
+                onAnswersChange(it.toList().map(AnswerInput::Index))
             }
         )
+        is QuestionDisplayType.Textarea -> TextareaAnswer(
+            onTextChange = {
+                onAnswersChange(listOf(AnswerInput.Content(index = 0, content = it)))
```

Comment:
Should the `Content` data class need an index field?

Reply:
Ok, we need it 😄 https://github.com/suho/kmm-ic/pull/120/files#diff-41fd63ac41b8847df57e438ab9f3c99dcb8353c1adc4bb57302d8618307ed8efR279-R283